macos/FluentUI/Button/Button.swift

```diff
@@ -433,11 +489,62 @@ open class Button: NSButton {
 			guard oldValue != style else {
 				return
 			}
+			setupBorderShadowsIfNeeded()
 			setColorValues(forStyle: style, accentColor: accentColor)
 			needsDisplay = true
 		}
 	}
 
+	var usesBorderShadows: Bool = false {
+		didSet {
+			guard oldValue != usesBorderShadows else {
+				return
+			}
+
+			if usesBorderShadows {
+				firstOuterDropShadowLayer = CALayer()
+				secondOuterDropShadowLayer = CALayer()
+				innerShadowLayer = CALayer()
+
+				// Ensure that the Shadow sublayers can in fact extend beyond the bounds of the Button View's
+				// layer so that the shadows don't get clipped
+				self.layer?.masksToBounds = false
+
+				if let firstOuterDropShadowLayer = firstOuterDropShadowLayer {
```

Comment:
These blocks are all identical minus some specific values. Thoughts on using a private enum ShadowLayerType which has shadowOffset/shadowRadius/shadowOpacity properties and then use a func which takes in a ShadowLayerType and executes one of these blocks?

Then this repetitive code becomes
```
update(shadowLayer: firstOuterDropShadow, type: .firstOuter)
update(shadowLayer: secondOuterDropShadow, type: .secondOuter)
update(shadowLayer:innerShadow, type: .inner)
```